### PR TITLE
Light fitting and light bulb/tubes refactor

### PR DESCRIPTION
**Pull request by coiax**
No user visible changes.

- Removed lightbulbs inherit the color, strength and range of their
fittings.
- Light fittings only break on mapload
- Light fittings now use Initialize() instead of spawns in new

- This'll allow for colored bulbs a lot easier, but doesn't introduce
any or change anything.

### CODE REVIEW

**Comment by Xhuis** [comment]
i don't know if it'll fit kyaa~

**Comment by ChangelingRain** [comment]
The bulb should determine the color of the light, not the fitting.

**Comment by coiax** [comment]
The bulb does.

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -476,12 +470,15 @@
 	var/obj/item/weapon/light/L = new light_type()
 	L.status = status
 	L.rigged = rigged
-	L.brightness = brightness
 
 	// light item inherits the switchcount, then zero it
 	L.switchcount = switchcount
 	switchcount = 0
 
+	L.l_range = l_range
```
Comment (ChangelingRain) [inline code/modules/power/lighting.dm]: uh??? this looks like removing the bulb changes the bulb's light vars????
It shouldn't do that

Comment (coiax) [inline code/modules/power/lighting.dm]: If you var edit the range of the light fitting, and then remove the bulb, the bulb inherits the light fitting's var editing

Comment (ChangelingRain) [inline code/modules/power/lighting.dm]: It shouldn't, you should edit the bulb's range instead

Comment (coiax) [inline code/modules/power/lighting.dm]: Fixtures don't store the bulbs, they store the bulb's type.

Comment (ChangelingRain) [inline code/modules/power/lighting.dm]: Then change that!

Comment (coiax) [inline code/modules/power/lighting.dm]: Okay.

**Comment by coiax** [comment]
Having to refactor all of lighting code to use actual bulb instances is taking a lot of work.

**Comment by coiax** [comment]
@ChangelingRain this is so much work for so little reward...

**Comment by KorPhaeron** [comment]
Reopen when this compiles, it's a good idea though

**Pull request closed without merging**